packages/webpack-loader/src/webpackLoader.ts

```diff
@@ -56,21 +56,18 @@ export function webpackLoader(
 
   EvalCache.clearForFile(this.resourcePath);
 
-  const resolveOptionsDefaults: webpack.ResolveOptions = {
-    conditionNames: ['require'],
-    extensions: ['.js', '.jsx', '.ts', '.tsx', '.json'],
-  };
   // ⚠ "this._compilation" limits loaders compatibility, however there seems to be no other way to access Webpack's
   // resolver.
   // There is this.resolve(), but it's asynchronous. Another option is to read the webpack.config.js, but it won't work
   // for programmatic usage. This API is used by many loaders/plugins, so hope we're safe for a while
   const resolveOptionsFromWebpackConfig: webpack.ResolveOptions = this._compilation?.options.resolve || {};
 
   const resolveSync = enhancedResolve.create.sync({
-    ...resolveOptionsDefaults,
     alias: resolveOptionsFromWebpackConfig.alias,
     modules: resolveOptionsFromWebpackConfig.modules,
     plugins: resolveOptionsFromWebpackConfig.plugins,
+    conditionNames: [...(resolveOptionsFromWebpackConfig.conditionNames ?? []), 'require'],
```

Comment:
Yeah I can see the problem you'd have with the setup you described in the PR description. However, changing the behaviour to do a complete inheritance by default of resolve options will be a breaking change for existing users. Here's an example: 

The current set of extensions include `.json`, if this is not configured in the webpack config - it can break when Linaria tries to resolve modules that import json files. In this case even though webpack does not configure `json`in resolve options, it will not fail because it will prefer to use `import`. However `Linaria` will then blow up because it can't resolve a JSON file

In this case I'd propose two alternatives to support your use case:

1. add a loader option to select what resolve options users would like to inherit:

`inheritResolveOptions: ['alias', 'modules', 'plugins', 'conditionNames]`

This way inheritance is possible without requiring a major version bump of the package

2. Allow users to create their own `resolveSync` function which will achieve the same with maybe greater control over the resolver use by Linaria. https://github.com/microsoft/griffel/blob/bd37c4fe0b21b33c1d548f7abf1b8c58dd1e5280/packages/webpack-loader/src/webpackLoader.ts#L59-L74

In addition to inheriting resolve options, you'd also need to configure the evaluation rules  so that you transform the TS code to CJS which is required by Linaria https://github.com/microsoft/griffel/blob/main/packages/webpack-loader/README.md#configuring-module-evaluation

Reply:
I've implemented the first suggestion.